### DISCUSSION TITLE
Directly save the excel file to local machine/server

### DISCUSSION THREAD

**Discussion by chiragthakuri** [PHPOffice/PhpSpreadsheet#3590] [Q&A]
Hello everyone,

I have been using this tool to create an excel file from a HTML table. It has worked well for me so far. However, right now I ran into an issue. Is it possible to directly generate an xlsx file onto the local machine folder/server (without the need for a download prompt)? 

I read the documentation and it says that if Sheet.js is executed from the browser directly, download prompt will appear. Is there anyway to circumvent this and directly create the folder in our local machine. I am using Sheet.js along with PHP and jQuery.

 Thank you!

**Comment by MarkBaker** [comment]
I would hope not!
Consider if it was possible to download a file directly without prompt. What folder would it download to? Would the code be able to say which folder it downloaded to? Would there be any security issues with downloading files to a local machine without a user being aware of it?

**Reply by chiragthakuri** [reply]
Oh yeah about security it is for a internal automation, not for general users.
 I need to setup a cron job that emails that excel file to our email address. 
Yes the code will specify a temp folder and delete the xlsx file after the email is sent!

**Comment by MarkBaker** [comment]
Your point about security being for internal automation is irrelevant. A browser should NEVER allow for the possibility of downloading a file without prompting the user.
If you need to setup a cron job, then you save the file on your server, just by specifying the path and filename, exactly as many of our example code snippets do, as the ['Hello World' example in the documentation does](https://phpspreadsheet.readthedocs.io/en/latest/#hello-world)

**Reply by chiragthakuri** [reply]
Thank you! This is what I was looking for!
